setup.py

```diff
@@ -1,10 +1,29 @@
-import os
 from setuptools import setup
-from setuptools.config import read_configuration
+import ast
+try:
+    from pathlib import Path
+except:
+    from pathlib2 import Path
 
-this_dir = os.path.dirname(__file__)
-cfg = read_configuration(os.path.join(this_dir, 'setup.cfg'))
-cfg["options"].update(cfg["metadata"])
-cfg = cfg["options"]
 
-setup(**cfg)
+def extract_global_vars(file):
+    t = file.read_text(encoding="utf-8")
+    m = ast.parse(t)
+    res = {}
+    for s in m.body:
+        if isinstance(s, ast.Assign):
+            vo = s.value
+            v = None
+            if isinstance(vo, ast.Str):
+                v = vo.s
+            elif isinstance(vo, ast.Num):
+                v = vo.n
+            
+            if v is not None:
+                for t in s.targets:
+                    if isinstance(t, ast.Name):
+                        res[t.id] = v
+    return res
+
+if __name__ == "__main__":
+    setup(version=extract_global_vars(Path(__file__).absolute().parent / "kaitaistruct.py")["__version__"])
```

Comment:
The `.absolute()` call shouldn't be necessary here - the code will work fine even if `__file__` is a relative path.

Reply:
it is jecessary, I had problems with other packages. Order matters.

Reply:
How would `.absolute()` make a difference though? If the input path is relative, `pathlib` will correctly return a new path (relative to the cwd) that points to kaitaistruct.py.

```python
>>> pathlib.Path("setup.py").parent / "kaitaistruct.py"                                         
PosixPath('kaitaistruct.py')
>>> pathlib.Path("../setup.py").parent / "kaitaistruct.py"                                      
PosixPath('../kaitaistruct.py')
>>> pathlib.Path("some/sub/dir/setup.py").parent / "kaitaistruct.py"                            
PosixPath('some/sub/dir/kaitaistruct.py')
>>> # Same with os.path
>>> os.path.join(os.path.dirname("setup.py"), "kaitaistruct.py")                                
'kaitaistruct.py'
>>> os.path.join(os.path.dirname("../setup.py"), "kaitaistruct.py")                             
'../kaitaistruct.py'
>>> os.path.join(os.path.dirname("some/sub/dir/setup.py"), "kaitaistruct.py")                   
'some/sub/dir/kaitaistruct.py'
```

Of course we're assuming that `__file__` is in fact relative to the current cwd, but if it isn't, `.absolute()´ wouldn't work correctly either.

Reply:
I remember that if my cwd is the parent dir of the parent dir of a file, without `absolute()` everything works incorrectly. =